common/include/joy.h

```diff
@@ -49,19 +50,19 @@ extern int event_joystick_get_button(const d_event &event);
 namespace dcx {
 
 #if DXX_MAX_BUTTONS_PER_JOYSTICK
-extern void joy_button_handler(SDL_JoyButtonEvent *jbe);
+extern window_event_result joy_button_handler(SDL_JoyButtonEvent *jbe);
```

Comment:
Fixed. I can confirm that `common/arch/sdl/event.cpp` and `common/arch/sdl/joy.cpp` now compile with `scons max_joysticks=0` and `scons max_axes_per_joystick=0` (tested setting `DXX_MAX_AXES_PER_JOYSTICK`, `DXX_MAX_HATS_PER_JOYSTICK` and `DXX_MAX_BUTTONS_PER_JOYSTICK` to `0` in `dxxsconf.h` also).

However, note that unification/master won't build with `scons max_joysticks=0` or `scons max_axes_per_joystick=0` (filing a separate issue #258).